hkmc2/shared/src/main/scala/hkmc2/semantics/ucs/Normalization.scala

```diff
@@ -180,9 +175,7 @@ class Normalization(elaborator: Elaborator)(using raise: Raise, ctx: Ctx):
       case split @ Split.Cons(head, tail) =>
         log(s"CASE Cons ${head.showDbg}")
         head match
-          case Branch(thatScrutineeVar, Pattern.Var(alias), continuation) =>
-            Split.Let(alias, thatScrutineeVar, rec(continuation))
-          case Branch(test, Pattern.Lit(Tree.BoolLit(true)), continuation) =>
+          case Branch(Term.Ref(_: TestSymbol), Pattern.Lit(Tree.BoolLit(true)), continuation) =>
```

Comment:
This logic looks quite weird and brittle. Why do you need to assume that some other part of the UCS put a `TestSymbol` there which is matched specifically against `true`? Why not represent things more cleanly as two different cases in internal UCS the representation itself?